### PR TITLE
move vat-ibc/vat-network from config to network-proposal.js

### LINKED ISSUE #7044: vaults bootstrap includes extra bundle: vat-network
## What is the Problem Being Solved?
vat-network is not being used in the vaults release. by removing it from the bootstrap we can avoid having to worry about it being durable or upgradeable. This allows us to defer #[5973](https://github.com/Agoric/agoric-sdk/pull/5973)

### PR DESCRIPTION
**Pull request by warner**
fix(vats): move vat-ibc/vat-network from config to network-proposal.js

Previously, certain configurations included vat-ibc and vat-network
code as part of bootstrap.

These two vats are not properly upgradable, and are not used in our
initial launch configuration. To minimize risk, we don't want unused
code running on-chain, however we have development and test
configurations which *do* need IBC support, so we don't want to delete
this code entirely.

This commit removes ibc/network from `config.bundles`, and instead
creates "proposals" for them (just like we do for Pegasus and the
Inter protocol). You can run `agoric run
packages/vats/scripts/init-network.js` to create both a set of bundles
for installation, and a proposal file to submit in a transaction. Once
executed, this will cause the target chain to create vat-network and
vat-ibc, and to wire them up just as if they were included at
bootstrap time.

closes #7044

### CODE REVIEW

**Comment by warner** [comment]
I haven't gotten all the tests to pass yet. Questions:

In `test-boot-config.js`, we have
```js

const NON_UPGRADEABLE_VATS = [
  // TODO(#6687): enforce vat-network
  // IDEA: move vat-network to a CoreEval proposal?
  // 'vat-network',
  // 'vat-ibc',
  // 'pegasus',
  // TODO(#6687): prune centralSupply from prod config
  // 'centralSupply',
  'mints',
  'sharing',
];
```

I think I should remove the first four lines (the first TODO through `vat-ibc`). Should I also remove the 'pegasus' line? I see the pegasus proposal line in `demo-proposals.json`, but I don't think it's included in the config this is testing.

In `test-vaults-integration` ("audit bootstrap exports"), I've removed a couple of exported objects that we aren't creating anymore (with ifaces of `ProtocolHandler`, `callbacks`, `listener`). But I wasn't sure how to change `maxExports` and `maxNonDurable`. Should I reduce these? The new values emitted during my test run were
```
  ✔ audit bootstrap exports (692ms)
    ℹ bootstrap exports: 2 durable 12 total
```

`test-demo-config.js` is failing during setup because it's trying to `createVatByName('network')`, even though the proposal that drives it ought to be citing a `bundleID`. I'm stll trying to figure this one out. The error output includes:

```
..
evaluateBundleCap {
  manifestBundleRef: { bundleName: 'coreProposal1_behaviors' },
  exportedGetManifest: 'getManifestForNetwork',
  vatAdminSvc: Promise { Object [Alleged: vatAdminService] {} }
}..

dm.invoke failed, informing calling vat (Error#1)
Error#1: unregistered bundle name 'network'
  at Object.invoke (.../swingset-vat/src/devices/vat-admin/device-vat-admin.js:283:7)
  at Object.invoke (.../swingset-vat/src/kernel/deviceManager.js:94:30)
  at invoke (.../swingset-vat/src/kernel/kernelSyscall.js:141:22)
  at kernelSyscallHandler (.../swingset-vat/src/kernel/kernelSyscall.js:234:8)
  at vatSyscallHandler (.../swingset-vat/src/kernel/kernel.js:1483:6)
  at syscallHandler (.../swingset-vat/src/kernel/vat-warehouse.js:42:12)
  at syscallFromWorker (.../swingset-vat/src/kernel/vat-loader/manager-helper.js:164:11)
  at doSyscall (.../swingset-vat/src/supervisors/supervisor-helper.js:72:3)
  at Object.callNow (.../swingset-vat/src/supervisors/supervisor-helper.js:114:1)
  at Proxy.eval (.../swingset-liveslots/src/liveslots.js:1036:19)
  at Object.createVatByName (.../swingset-vat/src/vats/vat-admin/vat-vat-admin.js:481:32)
  at Alleged: vatAdminService.method (.../exo/src/exo-tools.js:183:8)
  at Alleged: vatAdminService.In "createVatByName" method of (vatAdminService) [as createVatByName] (.../exo/src/exo-tools.js:42:14)
```

**Comment by dckc** [comment]
> In `test-boot-config.js`, we have
> 
> ```js
> const NON_UPGRADEABLE_VATS = [
>   // TODO(#6687): enforce vat-network
>   // IDEA: move vat-network to a CoreEval proposal?
>   // 'vat-network',
>   // 'vat-ibc',
>   // 'pegasus',
>   // TODO(#6687): prune centralSupply from prod config
>   // 'centralSupply',
>   'mints',
>   'sharing',
> ];
> ```
> 
> I think I should remove the first four lines (the first TODO through `vat-ibc`). Should I also remove the 'pegasus' line?

All 4 should be included, uncommented. The point here is to enforce that none of them is in a production config.

**Comment by dckc** [comment]
> In `test-vaults-integration` ("audit bootstrap exports"), ... I wasn't sure how to change `maxExports` and `maxNonDurable`. Should I reduce these?

Yes, please.

> The new values emitted during my test run were
> 
> ```
>   ✔ audit bootstrap exports (692ms)
>     ℹ bootstrap exports: 2 durable 12 total
> ```

File: packages/vats/scripts/init-network.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { makeHelpers } from '@agoric/deploy-script-support';
+
+export const defaultProposalBuilder = async ({ publishRef, install }) =>
```
Comment (dckc) [inline packages/vats/scripts/init-network.js:3]: I sure wish we had static types here... or some documentation on the shape of this thing. not critical.

I suppose it's an argument to `writeCoreProposal` which is a product of `makeHelpers`, which is imported. So there _is_ a follow-your-nose path, provided you're used to our define-before-use style.

File: packages/vats/src/core/basic-behaviors.js [hunk 2]
```diff
@@ -68,13 +68,25 @@ export const makeVatsFromBundles = async ({
...
+            name: vatName,
+          });
+          return vatInfo;
+        }
+        console.info(`createVat(${bundleID})`);
+        assert(bundleID);
+        const bcap = await E(svc).getBundleCap(bundleID);
         /** @type { Promise<CreateVatResults> } */
-        const vatInfo = E(svc).createVatByName(bundleName, {
+        const vatInfo = E(svc).createVat(bcap, {
```
Comment (dckc) [inline packages/vats/src/core/basic-behaviors.js:87]: nifty

File: packages/vats/src/core/chain-behaviors.js [hunk 3]
```diff
@@ -9,11 +9,6 @@ import {
   makeSubscriptionKit,
   observeIteration,
 } from '@agoric/notifier';
-import {
-  makeLoopbackProtocolHandler,
-  makeEchoConnectionHandler,
-  makeNonceMaker,
-} from '@agoric/swingset-vat/src/vats/network/index.js';
```
Comment (dckc) [inline packages/vats/src/core/chain-behaviors.js:16]: buh-bye :wave:

File: packages/vats/src/proposals/network-proposal.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
+// @ts-check
```
Comment (dckc) [inline packages/vats/src/proposals/network-proposal.js:1]: automated tests for this proposal would be great, but not likely cost-effective.

I saw this run on @warner 's desk. Here's hoping someone finds a moment to reproduce his results. But I don't think it's critical.

File: packages/vats/test/bootstrapTests/test-vaults-integration.js [hunk 5]
```diff
@@ -84,10 +84,6 @@ test('audit bootstrap exports', async t => {
       'prioritySenders manager': true,
       // TODO? move to provisioning vat?
       clientCreator: true,
-      // in registerNetworkProtocols(). TODO: outboard #7044
-      ProtocolHandler: true, // in makeLoopbackProtocolHandler()
```
Comment (dckc) [inline packages/vats/test/bootstrapTests/test-vaults-integration.js:88]: :clap:

**Review by dckc — APPROVED** [review]
right on target :dart:

File: packages/vats/src/proposals/network-proposal.js (hunk 4 above)
Comment (warner) [reply packages/vats/src/proposals/network-proposal.js:1]: FWIW the tests we ran were:
* use cosmic-swingset's `make scenario2-run-chain` and `make scenario2-run-client`
* run `agoric open --repl`, open a browser to the URL it prints
* in the repl, do and observe:
```
command[11]    E(E.get(home.ibcport)[0]).getLocalAddress()
history[11]    "/ibc-port/port-1"
```

**Comment by warner** [comment]
One other TODO that got resolved: in `test-boot-config.js`, the last test ("bootstrap permit visualization snapshot") uses t.snapshot on the output of a dependency analysis tool. This output changed, removing `space_networkVat` and `setupNetworkProtocols` as expected. But the snapshot also lost several lines about `space_zoe`, `space_provisioning`, and `space_bridgeManager`.

It turns out the analysis tool emits duplicate lines. Some of these duplicates went away as a result of the config changes, but not all of them. We aren't actually removing zoe.

cc @dckc when the dust settles, we might see if there's a way to have the tool not emit those duplicates, to reduce the false-positive rate on snapshot-changing commits like these.

**Comment by migueldingli1997** [comment]
Note to self -- I believe the command mentioned in the PR description is now: `agoric run packages/builders/scripts/vats/init-network.js`

**Pull request merged**